scripts/new-version-pr-generator.js

```diff
@@ -0,0 +1,193 @@
+const yaml = require('js-yaml');
+const semver = require('semver');
+
+const {version: PACKAGE_VERSION} = require('../package.json');
+const secrets = require('../secrets.json');
+const retry = require('./utilities/retry');
+
+const YARN_VERSION = yaml
+  .safeLoad(readFileSync(pathResolve(__dirname, '..', 'dev.yml'), 'utf8'))
+  .up.find((obj) => Object.keys(obj).includes('node')).node.yarn;
+
+// File paths and directories
+const root = pathResolve(__dirname, '../');
+const sandbox = pathResolve(root, 'sandbox');
+
+// Create the sandbox directory
+mkdir(sandbox);
+
+// Execute sync default options
+const execOpts = {stdio: 'inherit'};
+
+// Current release version
+const releaseVersion = `v${PACKAGE_VERSION}`;
+
+// Compute the base branch on polaris-styleguide (default: master)
+// Example: will open a PR against the v4 branch if the
+// version found in package.json is `4.0.0-beta.x`
+
+// Check if a version is a major pre-release:
+// isMajorPrerelease('2.0.0') => false
+// isMajorPrerelease('3.0.1-beta') => false
+// isMajorPrerelease('1.1.0-alpha') => false
+// isMajorPrerelease('2.0.0-beta.1') => true
+// isMajorPrerelease('3.0.0-alpha') => true
+function isMajorPrerelease(version) {
+  return (
+    semver.prerelease(version) &&
+    semver.minor(version) === 0 &&
+    semver.patch(version) === 0
+  );
+}
+
+const baseBranch = isMajorPrerelease(PACKAGE_VERSION)
+  ? `v${semver.major(PACKAGE_VERSION)}`
+  : 'master';
+
+// Polaris bot settings
+const polarisBotName = 'Shopify Polaris Bot';
+const polarisBotEmail = 'shopify-polaris-bot@users.noreply.github.com';
+const polarisBotToken = secrets.github['shopify-polaris'];
+const shopifyPolarisBotGitOverride = `GIT_COMMITTER_NAME='${polarisBotName}' GIT_COMMITTER_EMAIL='${polarisBotEmail}'`;
+
+const gitPolarisStyleguideDirectoryOverride = `--git-dir ${polarisStyleguide}/.git --work-tree=${polarisStyleguide}`;
+
+// Pull request template
+const updateBody = `
+## Version ${releaseVersion} of @shopify/polaris just got published!
+
+See what’s new: https://github.com/Shopify/polaris-react/releases/tag/${releaseVersion}
+
+cc @kaelig @dfmcphee @amrocha @BPScott
+
+---
+
+<details>
+<summary>🚨 What to do if you see “Your tests failed on CircleCI”?</summary>
+
+If tests fail, you may have to troubleshoot the problem locally.
+
+1. Checkout the \`update-polaris-${releaseVersion}\` branch:
+    \`\`\`bash
+    dev cd polaris-styleguide
+    git fetch
+    git checkout update-polaris-${releaseVersion}
+    dev up
+    \`\`\`
+1. Apply changes/fixes
+1. Commit
+1. Push your work to the \`update-polaris-${releaseVersion}\` branch:
+    \`\`\`bash
+    git push origin update-polaris-${releaseVersion}
+    \`\`\`
+1. Repeat until the build & tests go ✅
+1. 🎩
+1. :shipit:
+</details>
+`.trim();
+
+const updatePostObject = {
+  title: `Update @shopify/polaris to ${releaseVersion} 🚀`,
+  body: updateBody,
+  head: `update-polaris-${releaseVersion}`,
+  base: baseBranch,
+};
+
+ExecSyncQueue(
```

Comment:
Does this work? The function is called `execSyncQueue` without capitalization

Reply:
Nope I have cleaned this up, my bad!